### PR TITLE
Configurable logs & timeseries state retention

### PR DESCRIPTION
**Pull request by guillaumebort**
Refactoring how the TimeSeriesScheduler is instantiated (it was
a typeclass on Scheduling before but now we don't need to retrieve
the instance implicitely).

Also cleaning the code duplication between CuttleProject.start &
CuttleProject.build

### CODE REVIEW

File: core/src/main/scala/com/criteo/cuttle/Database.scala [hunk 1]
```diff
@@ -353,6 +354,15 @@ private[cuttle] case class Queries(logger: Logger) {
       (${id}, ${streams})
     """.update.run
 
+  def applyLogsRetention(logsRetention: ScalaDuration): ConnectionIO[Int] =
+    sql"""
```
Comment (dufrannea) [inline core/src/main/scala/com/criteo/cuttle/Database.scala]: Any reason why it is not written like this ?
```sql
DELETE es
FROM executions_streams es
JOIN executions e
ON es.id = e.id
WHERE end_time < ${Instant.now.minusSeconds(logsRetention.toSeconds)}
```

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala [hunk 2]
```diff
@@ -328,7 +325,7 @@ private[timeseries] object JobState {
   * The scheduler also allow to [[Backfill]] already computed partitions. The [[Backfill]] can be recursive
   * or not and an audit log of backfills is kept.
   */
-case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] {
+case class TimeSeriesScheduler(logger: Logger, stateRetention: Option[ScalaDuration] = None) extends Scheduler[TimeSeries] {
```
Comment (dufrannea) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala]: We should put a lower bound on this.

File: core/src/main/scala/com/criteo/cuttle/Database.scala (hunk 1 above)
Comment (guillaumebort) [reply core/src/main/scala/com/criteo/cuttle/Database.scala]: I tried but it seems not supported on Mysql. I can check again.

Comment (guillaumebort) [reply core/src/main/scala/com/criteo/cuttle/Database.scala]: It seems to work indeed.

**Pull request merged**